lib/arjdbc/postgresql/adapter.rb

```diff
@@ -52,8 +52,15 @@ def postgresql_version
       @postgresql_version ||=
         begin
           version = @connection.database_product
-          if version =~ /PostgreSQL (\d+)\.(\d+)\.(\d+)/
-            ($1.to_i * 10000) + ($2.to_i * 100) + $3.to_i
+          if version.match /PostgreSQL ((\d+\.)+\d+)/
+            version_numbers = $1.split('.').map(&:to_i)
+            # PostgreSQL version representation does not have more than 3 digits
+            return 0 if version_numbers.length > 3
```

Comment:
There are servers with 4 version parts, e.g. some EnterpriseDB distributions. The JDBC driver just ignores the 4th part, see `pgjdbc/src/main/java/org/postgresql/core/ServerVersion.java`. It's referenced with a bug: https://github.com/pgjdbc/pgjdbc/issues/667. ARJDBC should ignore a 4th part as well (if there is one).